tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/ProfilerSkipped/ProfilerSkippedMethodIntegration.cs

```diff
@@ -0,0 +1,82 @@
+// <copyright file="ProfilerSkippedMethodIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.ComponentModel;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.ContinuousProfiler;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.ProfilerSkipped
+{
+    /// <summary>
+    /// Calltarget instrumentation to skip profiler sampling during method execution
+    /// </summary>
+    [Browsable(false)]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class ProfilerSkippedMethodIntegration
+    {
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+        /// <param name="methodHandle">The RuntimeMethodHandle representing the instrumented method</param>
+        /// <param name="typeHandle">The RuntimeTypeHandle representing the instrumented method's owning type</param>
+        /// <returns>Calltarget state value</returns>
+        internal static CallTargetState OnMethodBegin<TTarget>(TTarget instance, RuntimeMethodHandle methodHandle, RuntimeTypeHandle typeHandle)
+        {
+            Profiler.Instance.ContextTracker.SetThreadMetaInfo(LockStatus.Lock);
```

Comment:
I don't really see a lot of instances of `Profiler.Instance.*` in the codebase, so just commenting as this is new/strange to me.

Do we need to check `Profiler.Instance.ContextTracker.IsEnabled` prior to this?